File: API/Entities/Entry.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace API.Entities;

public class Entry
{
    public int Id { get; set; }
    public string Content { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public string? MediaUrls { get; set; }

    // Foreign keys
    public int DiaryId { get; set; }

    // Navigation properties
    public Diary Diary { get; set; } = null!;
}

File: client/src/types/entry.ts
export interface Entry {
    id: number;
    content: string;
    createAt: string;
    updatedAt: string;
    mediaUrls?: string[];
    diaryId: number;
}

export interface CreateEntry {
    content: string;
    mediaUrls?: string[];
}

export interface UpdateEntry {
    content?: string;
    mediaUrls?: string[];
}